src/Sylius/Component/Core/OrderProcessing/OrderPaymentProcessor.php

```diff
@@ -72,7 +72,7 @@ private function setPaymentMethodIfNeeded(OrderInterface $order, PaymentInterfac
     {
         $lastPayment = $this->getLastPayment($order);
 
-        if (null === $lastPayment) {
+        if (false === $lastPayment) {
```

Comment:
`::getLastPayment` should return null in similar fashion like it's done in the `Order` model